File: operations/src/main/java/de/davelee/trams/operations/request/SellVehicleRequest.java
package de.davelee.trams.operations.request;

import lombok.*;

/**
 * This class is part of the TraMS Operations REST API. It represents a request to sell a vehicle for the particular
 * company fulfilling the details supplied.
 * @author Dave Lee
 */
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Setter
public class SellVehicleRequest {

    /**
     * The fleet number of the vehicle to sell.
     */
    private String fleetNumber;

    /**
     * The company that owns this vehicle at the moment.
     */
    private String company;

}

File: desktop/src/app/shared/sellvehicle.request.ts
/**
 * This class defines a model for requests to sell vehicles in TraMS.
 */
export class SellVehicleRequest {

    private company: string;
    private fleetNumber: string;

    /**
     * Construct a new request to sell a vehicle.
     * @param company the company that the vehicle belonged to.
     * @param fleetNumber the fleet number of the vehicle to sell.
     */

    constructor(company: string, fleetNumber: string) {
        this.company = company;
        this.fleetNumber = fleetNumber;
    }
}